day1/day1.py

```diff
@@ -0,0 +1,39 @@
+def list_changes(data):
+    changes = []
+    for line in data:
+        line = line.strip()
+        if (line[0] == '+'):
+            changes.append(int(line[1:]))
+        else:
+            changes.append( - int(line[1:]))
+    return changes
+
+
+def first_repeat(data):
+    changes = list_changes(data)
+    curr_freq = 0
+    i = 0
+    freqs = []
+    while True:
+        next_item = changes[i % len(changes)]
+        curr_freq += next_item
+        if curr_freq in freqs:
+            return curr_freq 
+        freqs.append(curr_freq)
+        i += 1
+
+
+
+def final_freq(data):
+    changes = list_changes(data)
+    curr_freq = 0
+    for change in changes:
+        curr_freq += change
+    return curr_freq 
```

Comment:
You can approach it in a more declarative way -> instead of doing for, you can write it using map/reduce (http://book.pythontips.com/en/latest/map_filter.html)